src/catchup/ApplyBucketsWork.cpp

```diff
@@ -21,25 +21,24 @@ namespace stellar
 ApplyBucketsWork::ApplyBucketsWork(
     Application& app, WorkParent& parent,
     std::map<std::string, std::shared_ptr<Bucket>>& buckets,
-    HistoryArchiveState& applyState,
-    LedgerHeaderHistoryEntry const& firstVerified)
+    HistoryArchiveState applyState, LedgerHeaderHistoryEntry applyAt)
     : Work(app, parent, std::string("apply-buckets"))
     , mBuckets(buckets)
-    , mApplyState(applyState)
-    , mFirstVerified(firstVerified)
+    , mApplyState(std::move(applyState))
+    , mApplyAt{std::move(applyAt)}
```

Comment:
why ditching the mFirstVerified name?  Maybe it's not the best name but mApplyAt seems to convey less information. Also you missed a few references to the old name (firstVerified)